evm/gas-service.js

```diff
@@ -0,0 +1,414 @@
+'use strict';
+
+const { ethers } = require('hardhat');
+const {
+    getDefaultProvider,
+    utils: { formatEther, parseEther },
+    Contract,
+    BigNumber,
+} = ethers;
+const { Command, Option } = require('commander');
+const {
+    printInfo,
+    printWalletInfo,
+    printWarn,
+    mainProcessor,
+    prompt,
+    getGasOptions,
+    wasEventEmitted,
+    isValidAddress,
+    validateParameters,
+} = require('./utils');
+const { addBaseOptions } = require('./cli-utils');
+const IAxelarGasService = require('@axelar-network/axelar-gmp-sdk-solidity/interfaces/IAxelarGasService.json');
+const { getWallet } = require('./sign-utils');
+
+async function getGasInfo(env, chain, destinationChains) {
+    const apiUrl = env === 'testnet' ? 'https://testnet.api.axelarscan.io/gmp/getFees' : 'https://api.axelarscan.io/gmp/getFees';
+
+    return Promise.all(
+        destinationChains.map((destinationChain) => {
+            let feeType = 0;
+            let blobBaseFee = 0;
+
+            switch (destinationChain.toLowerCase()) {
+                case 'ethereum-sepolia':
+
+                // eslint-disable-next-line no-fallthrough
+                case 'ethereum': {
+                    feeType = 0;
+                    blobBaseFee = env === 'testnet' ? 50187563959 : 1;
+                    break;
+                }
+
+                case 'base':
+
+                // eslint-disable-next-line no-fallthrough
+                case 'optimism': {
+                    feeType = 1; // Optimism Ecotone
+                    blobBaseFee = 0;
+                    break;
+                }
+            }
+
+            return fetch(apiUrl, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                    sourceChain: chain.axelarId,
+                    destinationChain,
+                    sourceTokenAddress: '0x0000000000000000000000000000000000000000',
+                }),
+            })
+                .then((res) => res.json())
+                .then(({ result }) => {
+                    const {
+                        source_base_fee: sourceBaseFee,
+                        source_token: {
+                            gas_price_in_units: { value: gasPrice },
+                            decimals,
+                        },
+                        execute_gas_multiplier: multiplier = 1.1,
+                    } = result;
+
+                    const axelarBaseFee = Math.ceil(parseFloat(sourceBaseFee) * Math.pow(10, decimals));
+                    const relativeGasPrice = Math.ceil(parseFloat(gasPrice) * parseFloat(multiplier));
+
+                    return [feeType, axelarBaseFee, relativeGasPrice, blobBaseFee * relativeGasPrice];
```

Comment:
```suggestion
                    return [feeType, axelarBaseFee, relativeGasPrice, blobBaseFee * destinationTokenPrice / srcTokenPrice];
```
blobBaseFee itself is the blob gas price, e.g. 1 gwei, so we just need to normalize it, and not multiply regular gas price/base fee